pkg/apis/core/v1alpha1/types.go

```diff
@@ -56,6 +58,46 @@ type DataVolumeSpec struct {
 	Preallocation *bool `json:"preallocation,omitempty"`
 }
 
+// StorageSpec defines the Storage type specification
+type StorageSpec struct {
```

Comment:
If this has exactly the same members as `PersistentVolumeClaimSpec` why not:

`type StorageSpec corev1.PersistentVolumeClaimSpec`

Reply:
The story is long, it started like you propose (or similar, because the corev1.PersistentVolumeClaimSpec in that first version was embeded), but then I needed to add some fields, and decided I might need to remove some PVC fields, or change a description.

I think this API needs some better documentation in the struct so I'll leave it as is, and improve the inline descriptions.

Reply:
In the meantime all the fields I have added where removed ;) so now it looks like PVCSpec, but I plan on improving the documentation, and I thin being a little bit independent from PVC is good. 

If think my arguments are too weak, I can change this.

Reply:
Its an array and has omitempty. In fact all fields on PVC are optional in schema ("// +optional. and omitempty") Some are only enforced by Webhooks/Validators - including resources.requests[storage] and accessModes.